### PR TITLE
inject global mixins into index.d.ts

### PR DESCRIPTION
**Pull request by Zyie**
##### Description of change

So here is a pretty basic solution for #6993 

~~All it does is append the `GlobalMixins` if it exists in `global.d.ts` to the `index.d.ts` file. 
Seems to work pretty well but feel free to close this if you guys come up with a better solution~~

New approach, now just injects a path that points to the `global.d.ts` file

example layout
```
/// <reference path="./global.d.ts" />
```

### CODE REVIEW

**Review by ivanpopelyshev — APPROVED** [review]

**Comment by bigtimebuddy** [comment]
Yes!!! Thank you.

This feels like a bit of a hack, but good enough to unblock us. Couple of small organizational bits, could you:

* move this to a folder at the root level called scripts. The tools folder is for package-structured things.
* can you add some comments in your script about why this script is necessary? Having some comments like that are helpful so we don't have to dig thru GitHub later. Once api-extractor support this, we'd remove the hack.
* make sure eslint is considering this script.
* optionally, we could use ts-node and write the script in TS. Would be a nice, but unnecessary. Up to you if you wanna do this.

**Comment by Zyie** [comment]
> This feels like a bit of a hack,
Yes definitely is a hack 😄 

> * optionally, we could use ts-node and write the script in TS. Would be a nice, but unnecessary. Up to you if you wanna do this.
Yeah i dont mind doing this!

**Comment by bigtimebuddy** [comment]
Thanks @Zyie, you're my hero.

**Comment by Zyie** [comment]
@bigtimebuddy updated!

**Comment by bigtimebuddy** [comment]
Could you also check the bundles typings? Not sure how we might Mixin those global there too.

**Comment by Zyie** [comment]
@bigtimebuddy ok so did a quick test with `pixi.js` and `pixi.js-legacy`, seems to be working now

all we do is bundle all the global mixins together and add them to the `index.d.ts`

also `canvas-renderer` had some extra declarations in its `global.d.ts` which has now been handled!

**Review by bigtimebuddy — APPROVED** [review]
Code looks good, Will try to give this a spin locally. Thanks for handling the feedback.

**Comment by bigtimebuddy** [comment]
@SukantPal do you have any feedback on this approach?

**Comment by SukantPal** [comment]
@bigtimebuddy Why are we appending `global.d.ts` when `///<reference path="./global.d.ts" />` seems to work? Why not just add the reference to the `index.d.ts` in the script? We will have to edit this script whenever `global.d.ts` breaks the definitions (which we may not even notice), as in the case of the canvas package.

**Comment by Zyie** [comment]
> Why are we appending `global.d.ts` when `///<reference path="./global.d.ts" />` seems to work? Why not just add the reference to the `index.d.ts` in the script?

good point. I just did a test with adding

```
/// <reference path="../@pixi/display/global.d.ts" />
/// <reference path="../@pixi/mixin-cache-as-bitmap/global.d.ts" />
```
inside the `index.d.ts` of `pixi.js` and it works!

I'll update this PR to do this instead, its much simpler and less error prone

**Comment by SukantPal** [comment]
@Zyie Could we add this into `@pixi-build-tools/lerna-monorepo-api-extractor`?

https://github.com/SukantPal/pixi-build-tools/tree/master/packages/api-extractor-lerna-monorepo

**Comment by SukantPal** [comment]
If you want, we can bring that into the pixijs org @bigtimebuddy

**Comment by Zyie** [comment]
@SukantPal @bigtimebuddy updated the code to use the path reference instead.

I'll move it over to the build tool if @bigtimebuddy is happy with that, just let me know!

**Comment by bigtimebuddy** [comment]
I think we should try to get this in as-is. And then figure if it makes sense to add to build-tools. Ultimately this seems like something specific to this project but if there's a way to handle it in a general or explicit/config way, what's be good. I'll leave that up to @SukantPal to determine.

**Comment by SukantPal** [comment]
@Zyie We can merge this and I'll do stuff in pixi-build-tools later

**Review by SukantPal — APPROVED** [review]

File: scripts/injectGlobalMixins.ts
Comment (bigtimebuddy) [inline scripts/injectGlobalMixins.ts:79]: Can you see if you can use `/// <reference types="${pkg.name}/global.d.ts" />`? It would be nice to avoid any relative pathing issues (for instance symlinked packages) and using node's resolve instead. Not sure if this works tho.

Comment (Zyie) [reply scripts/injectGlobalMixins.ts:79]: ahhhh!!

this didn't work, however I have found this https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html

so we can do `/// <reference types="@pixi/core" />` inside `pixi.js`

Comment (Zyie) [reply scripts/injectGlobalMixins.ts:79]: @bigtimebuddy updated!

**Comment by bigtimebuddy** [comment]
Thank you @Zyie!

**Pull request merged**
